internal/hardware/validator.go

```diff
@@ -576,3 +575,44 @@ func compileDiskReasonTemplate(template string, wildcards ...interface{}) *regex
 	}
 	return tmp
 }
+
+// preciseHumanizeBytes provides more precise disk size formatting for error messages
+// to avoid confusion when operator additions create requirements like 107GB vs user's 100GB
+// IMPORTANT: Rounds UP decimal values to ensure users provide sufficient disk space
+func preciseHumanizeBytes(bytes uint64) string {
+	const (
+		KB = 1000
+		MB = 1000 * KB
+		GB = 1000 * MB
+		TB = 1000 * GB
+	)
+
+	switch {
+	case bytes >= TB:
+		tb := float64(bytes) / float64(TB)
+		if tb == float64(int64(tb)) {
+			return fmt.Sprintf("%d TB", int64(tb))
+		}
+		return fmt.Sprintf("%.1f TB", math.Ceil(tb*10)/10)
```

Comment:
this block of code is repeated 4 times, can we factorize it?
this can be a way: https://yourbasic.org/golang/formatting-byte-size-to-human-readable-format/
or this library: https://github.com/c2h5oh/datasize